rubicon/objc/api.py

```diff
@@ -803,36 +795,21 @@ def __getattr__(self, name):
 
         if method:
             return ObjCBoundMethod(method, self)
-        else:
+
+        # Check if the attribute name corresponds to an instance attribute defined at
+        # runtime from Python. Run it if available, otherwise raise an AttributeError.
+        pyo_wrapper = libobjc.objc_getAssociatedObject(self, SEL(name))
+
+        if pyo_wrapper.value is None:
             raise AttributeError('%s.%s %s has no attribute %s' % (
                 type(self).__module__, type(self).__qualname__, self.objc_class.name, name)
             )
+        address = get_ivar(pyo_wrapper, "wrapped_pointer")
+        pyo = cast(address.value, py_object)
 
-    def _ensure_keep_alive(self) -> None:
-        """Ensure that this :class:`ObjCInstance` is kept alive as long as its Objective-C object exists."""
-
-        # Check first if this ObjCInstance is already kept alive,
-        # in which case we don't need to do anything.
-        if self.ptr.value not in ObjCInstance._keep_alive_objects:
-            ObjCInstance._keep_alive_objects[self.ptr.value] = self
-
-            # Create a DeallocationObserver and associate it with the Objective-C object.
-            # When the Objective-C object is deallocated, the observer will remove
-            # this ObjCInstance from the _keep_alive_objects dictionary.
-            # This will cause the ObjCInstance to be destroyed,
-            # assuming that there are no other references to it from Python
-            # (which shouldn't happen - if Python code has a reference to an ObjCInstance,
-            # it's expected to also retain a reference to the Objective-C object).
-            observer = send_message(
-                send_message(get_class('DeallocationObserver'), 'alloc', restype=objc_id, argtypes=[]),
-                'initWithObject:', self, restype=objc_id, argtypes=[objc_id]
-            )
-            libobjc.objc_setAssociatedObject(self, observer, observer, 0x301)
-            # The observer is now retained by the object we associate it to.
-            # We now release our own reference to the observer,
-            # so that the only remaining reference is released
-            # when the associated object is deallocated.
-            send_message(observer, 'release', restype=None, argtypes=[])
+        send_message(pyo_wrapper, 'release', restype=None, argtypes=[])
```

Comment:
Never mind, the explicit call to release is not necessary after all. It appears that the object returned by `libobjc.objc_getAssociatedObject()` is autoreleased. Since the tests don't use an `NSAutoreleasePool`, those references where never cleaned up.